packages/twenty-front/src/modules/workflow/hooks/usePopulateStepsOutputSchema.ts

```diff
@@ -0,0 +1,68 @@
+import { useRecoilComponentCallbackStateV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentCallbackStateV2';
+import { stepsOutputSchemaComponentFamilyState } from '@/workflow/states/stepsOutputSchemaFamilyState';
+import { WorkflowVersion } from '@/workflow/types/Workflow';
+import { splitWorkflowTriggerEventName } from '@/workflow/utils/splitWorkflowTriggerEventName';
+import { getActionIcon } from '@/workflow/workflow-steps/workflow-actions/utils/getActionIcon';
+import { getTriggerIcon } from '@/workflow/workflow-trigger/utils/getTriggerIcon';
+import {
+  OutputSchema,
+  StepOutputSchema,
+} from '@/workflow/workflow-variables/types/StepOutputSchema';
+import { getTriggerStepName } from '@/workflow/workflow-variables/utils/getTriggerStepName';
+import { useRecoilCallback } from 'recoil';
+import { isDefined } from 'twenty-shared';
+
+export const usePopulateStepsOutputSchema = (workflowVersionId: string) => {
+  const stepsOutputSchemaFamilyState = useRecoilComponentCallbackStateV2(
+    stepsOutputSchemaComponentFamilyState,
+    workflowVersionId,
+  );
+
+  const populateStepsOutputSchema = useRecoilCallback(
+    ({ set }) =>
+      (workflowVersion: WorkflowVersion) => {
+        workflowVersion?.steps?.forEach((step) => {
+          const stepOutputSchema: StepOutputSchema = {
+            id: step.id,
+            name: step.name,
+            icon: getActionIcon(step.type),
+            outputSchema: step.settings.outputSchema as OutputSchema,
+          };
+
+          set(stepsOutputSchemaFamilyState(step.id), stepOutputSchema);
+        });
+
+        const trigger = workflowVersion.trigger;
+
+        if (isDefined(trigger)) {
+          const triggerIconKey =
+            trigger.type === 'DATABASE_EVENT'
+              ? getTriggerIcon({
+                  type: trigger.type,
+                  eventName: splitWorkflowTriggerEventName(
+                    trigger.settings?.eventName,
+                  ).event,
+                })
+              : getTriggerIcon({
+                  type: trigger.type,
+                });
+
+          const triggerOutputSchema: StepOutputSchema = {
+            id: 'trigger',
```

Comment:
You can use the `TRIGGER_STEP_ID` constant.

```suggestion
            id: TRIGGER_STEP_ID,
```